nixos/modules/services/torrent/qbittorrent-nox.nix

```
@@ -0,0 +1,108 @@
+{ config, lib, pkgs, ... }:
+let
+  cfg = config.services.qbittorrent-nox;
+  UID = 888;
+  GID = 888;
+in
+{
+  options.services.qbittorrent-nox = {
+    enable = lib.mkEnableOption (lib.mdDoc "qBittorrent-nox headless");
+
+    dataDir = lib.mkOption {
+      type = lib.types.path;
+      default = "/var/lib/qbittorrent-nox";
+      description = lib.mdDoc ''
+        The directory where qBittorrent-nox stores its data files.
+      '';
+    };
+
+    user = lib.mkOption {
+      type = lib.types.str;
+      default = "qbittorrent-nox";
+      description = lib.mdDoc ''
+        User account under which qBittorrent-nox runs.
+      '';
+    };
+
+    group = lib.mkOption {
+      type = lib.types.str;
+      default = "qbittorrent-nox";
+      description = lib.mdDoc ''
+        Group under which qBittorrent-nox runs.
+      '';
+    };
+
+    port = lib.mkOption {
+      type = lib.types.port;
+      default = 8080;
+      description = lib.mdDoc ''
+        qBittorrent-nox web UI port.
+      '';
+    };
+
+    openFirewall = lib.mkOption {
+      type = lib.types.bool;
+      default = false;
+      description = lib.mdDoc ''
+        Open services.qbittorrent-nox.port to the outside network.
+      '';
+    };
+
+    package = lib.mkOption {
+      type = lib.types.package;
+      default = pkgs.qbittorrent-nox;
+      defaultText = lib.literalExpression "pkgs.qbittorrent-nox";
+      description = lib.mdDoc ''
+        The qBittorrent-nox package to use.
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    networking.firewall = lib.mkIf cfg.openFirewall {
+      allowedTCPPorts = [ cfg.port ];
+    };
+
+    systemd = {
+      services.qbittorrent-nox = {
+        description = "qBittorrent-nox service";
+        documentation = [ "man:qbittorrent-nox(1)" ];
+        after = [ "network.target" ];
+        wantedBy = [ "multi-user.target" ];
+
+        serviceConfig = {
+          Type = "simple";
+          User = cfg.user;
+          Group = cfg.group;
+
+          ExecStart = "${cfg.package}/bin/qbittorrent-nox";
+          # To prevent "Quit & shutdown daemon" from working; we want systemd to manage it!
+          Restart = "on-success";
+          UMask = "0022";
+        };
+        environment.QBT_PROFILE = cfg.dataDir; # required
+      };
+
+      tmpfiles.settings."qbittorrent-nox" = {
+        "${cfg.dataDir}" = {
+          d = {
+            user = cfg.user;
+            group = cfg.group;
+            mode = "0755";
+          };
+        };
+      };
+    };
+
+    users.users = lib.mkIf (cfg.user == "qbittorrent-nox") {
+      qbittorrent-nox = {
+        group = cfg.group;
+        uid = UID;
+      };
+    };
+
+    users.groups = lib.mkIf (cfg.group == "qbittorrent-nox") {
+      qbittorrent-nox = { gid = GID; };
+    };
+  };
+}
```

A doc would be great
```suggestion
  meta.doc = ./qbittorrent-nox.md;
}
```

---

Where does this doc file go? Can I just inline it?

---

> Where does this doc file go? Can I just inline it?

`./qbittorrent-nox.md` here means `nixos/modules/services/torrent/qbittorrent-nox.md`

---

waiting on final approval to write docs now.